fastlane/Fastfile

```diff
@@ -58,37 +72,103 @@ lane :release do |options|
   #pod_lib_lint(podspec: "StreamChat.podspec", allow_warnings: true)
   #pod_lib_lint(podspec: "StreamChatUI.podspec", allow_warnings: true)
 
+  # Bump Podspec Version
   version_bump_podspec(path: "StreamChat.podspec", version_number: version_number)
   version_bump_podspec(path: "StreamChatUI.podspec", version_number: version_number)
 
+  # Bump XCFramework Version
+  version_bump_podspec(path: "StreamChat-XCFramework.podspec", version_number: version_number)
+  version_bump_podspec(path: "StreamChatUI-XCFramework.podspec", version_number: version_number)
+
+  # Stage all changes
   sh("git add -A")
 
   if(!prompt(text: "Will commit changes. All looking good?", boolean: true))
     UI.user_error!("Not committing changes")
   end
 
   sh("git commit -m 'Bump #{version_number}'")
-  sh("git tag #{version_number}")
 
   if(!prompt(text: "Will push changes. All looking good?", boolean: true))
     UI.user_error!("Not pushing changes")
   end
 
-  push_to_git_remote(tags: true)
+  push_to_git_remote(tags: false)
+
+  # Open a new PR
+  create_pull_request(
+    api_token: ENV["GITHUB_TOKEN"],
+    repo: "GetStream/stream-chat-swift",
+    title: "#{version_number} Release",
+    head: "release/#{version_number}",
+    base: "main",
+    body: "#{changes}"
+)
+
+  UI.success("Successfully started release #{version_number}")
+end
+
+desc "Completes an SDK Release"
+lane :publish_release do |options|
+
+  version = options[:version]
+
+  # Make xcframeworks
+  make frameworks
+
+  # Update SPM Repo
+  sh("git clone git@github.com:GetStream/stream-chat-swift-spm.git ../StreamSPM")
+  file = File.open("../StreamSPM/Package.swift")
+  file_data = file.read
+
+  File.open("../StreamSPM/Package.swift").each do |line|
+  if line.include?("url") && line.include?("StreamChat.zip")
+      streamChat = "url: \"https://github.com/GetStream/stream-chat-swift/releases/download/#{version}/StreamChat.zip\","
+      file_data.gsub!(line, streamChat)
+  end
+  if line.include?("url") && line.include?("StreamChatUI.zip")
+      streamChatUI = "url: \"https://github.com/GetStream/stream-chat-swift/releases/download/#{version}/StreamChatUI.zip\","
+      file_data.gsub!(line, streamChatUI)
+  end
+end
 
+  # Write the new changes
+  File.open("../StreamSPM/Package.swift", "w") { |file| file << file_data }
+
+  # Update the repo
+  sh("cd ../StreamSPM && git add -A")
+  sh("git commit -m 'Bump #{version}'")
+  sh("git tag #{version}")
+  sh("git push origin main --force --tags")
+
+  # Clean Up
+  sh("cd ../")
+  sh("rm -r StreamSPM")
+
+  # Generate Checksums
+  sh("swift package compute-checksum Products/StreamChat.zip")
+  sh("swift package compute-checksum Products/StreamChatUI.zip")
```

Comment:
The returned checksum needs to be reapplied to the `../StreamSPM/Package.swift` file

Reply:
Yeah exactly, currently working on this one 💯